packages/backend/src/program/application/programService.ts

```diff
@@ -6,13 +6,14 @@ import { QuestionnaireData } from '../domain/types'
 
 import ProgramsJson from '../infrastructure/programsJson'
 import { currentDateService } from '../infrastructure/currentDate'
-import { publicodesService } from '../infrastructure/publicodes'
+import { PublicodesService } from '../infrastructure/publicodesService'
 
 export default class ProgramService {
   private _program: ProgramFeatures
 
   constructor() {
-    this._program = new ProgramFeatures(ProgramsJson.getInstance(), currentDateService, publicodesService)
+    const programsService = ProgramsJson.getInstance()
+    this._program = new ProgramFeatures(programsService, currentDateService, new PublicodesService(programsService.getAll()))
```

Comment:
:thinking: : Je me demande si this._program contient aussi les programmes qui ne sont plus valide ? (pour les appels depuis le front)